version: "0.1"
services:
  transcendence-app:
    restart: always
    depends_on:
      - "db"
    build:
      context: ./transcendence-app
      target: development
      args:
        - TRANSCENDENCE_APP_DATA=${TRANSCENDENCE_APP_DATA:-/var/lib/transcendence-app}
    volumes:
      - transcendence-app-data:${TRANSCENDENCE_APP_DATA:-/var/lib/transcendence-app}
      - ./transcendence-app:/transcendence-app
      - /transcendence-app/node_modules
    environment:
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=postgres
      - FORTYTWO_APP_CALLBACK_URL=http://localhost:3000/api/v1/auth/login
      - FORTYTWO_APP_AUTHORIZATION_URL=https://api.intra.42.fr/oauth/authorize
      - FORTYTWO_APP_TOKEN_URL=https://api.intra.42.fr/oauth/token
      - FORTYTWO_APP_PROFILE_URL=https://api.intra.42.fr/v2/me
      - MEMCACHED_HOST=memcached
      - MEMCACHED_PORT=11211
      - POSTGRES_HOST=db
      - POSTGRES_PORT=5432
      - POSTGRES_DB=ft_transcendence

  db:
    image: postgres:14.4-alpine3.16
    restart: always
    user: postgres
    volumes:
      - ./db/fill-db.sh:/docker-entrypoint-initdb.d/fill-db.sh
      - ./db/setup/:/var/lib/postgresql/setup/
    environment:
      - POSTGRES_DB=ft_transcendence
volumes:
  transcendence-app-data:
